### PR TITLE
Updates godocs for trustedCA field of proxy api type

### PR DESCRIPTION
**Pull request by danehans**
Updates proxy trustedCA field godocs to provide additional explanation.

/assign @bparees @abhinavdahiya

### CODE REVIEW

File: config/v1/types_proxy.go [hunk 1]
```diff
@@ -42,17 +42,18 @@ type ProxySpec struct {
...
 	// trustedCA is a reference to a ConfigMap containing a CA certificate bundle used
 	// for client egress HTTPS connections. The certificate bundle must be from the CA
-	// that signed the proxy's certificate and be signed for everything. trustedCA should
-	// only be consumed by a proxy validator. The validator is responsible for reading
-	// ConfigMapNameReference, validating the certificate and copying "ca-bundle.crt"
-	// from data to a ConfigMap in the namespace of an operator configured for proxy.
-	// The namespace for this ConfigMap is "openshift-config". Here is an example
+	// that signed the proxy's certificate and be signed for everything. The trustedCA
+	// field should only be consumed by a proxy validator. The validator is responsible
+	// for reading the ConfigMap referenced by ConfigMapNameReference, validating the
```
Comment (bparees) [inline config/v1/types_proxy.go]:
```suggestion
	// for reading the ConfigMap referenced by trustedCA, validating the
```

File: config/v1/types_proxy.go [hunk 2]
```diff
@@ -42,17 +42,18 @@ type ProxySpec struct {
...
+	// certificate bundle and copying the bundle from "ca-bundle.crt" to a ConfigMap
```
Comment (bparees) [inline config/v1/types_proxy.go]: something in this doc should make it clear that the configmap referenced by trustedCA must contain a key named `ca-bundle.crt` as that is the only key the validator+publisher are going to care about.

**Comment by danehans** [comment]
@bparees I addressed your comments in my latest commit.

**Pull request merged**